Clojure/Clojure.sublime-syntax

```diff
@@ -1,1123 +1,269 @@
-        - match: '(?=\])'
-          pop: true
-        - include: binding
-    - match: (?=#?\()
-      push:
-        - meta_scope: meta.structure.binding_exp.sexp.clojure
-        - match: '(?=\])'
-          pop: true
-        - match: (?=#?\()
-          push:
-            - match: (?<=\))
-              pop: true
-            - include: function
-            - include: function_multi_method
-            - include: lambda
-            - include: macro
-            - include: sexpr
-        - include: binding
-    - match: '\['
-      scope: punctuation.definition.vector.begin.clojure
-      push:
-        - meta_scope: meta.structure.binding_exp.vector.clojure
-        - match: '(?=\])'
-          pop: true
-        - match: '(?<=\[)'
-          comment: "TODO: merge with vector"
-          push:
-            - meta_scope: meta.expression.vector.clojure
-            - match: '\]'
-              scope: punctuation.definition.vector.end.clojure
-              pop: true
-            - include: all
-        - include: binding
-    - match: '\{'
-      scope: punctuation.definition.map.begin.clojure
-      push:
-        - meta_scope: meta.structure.binding_exp.map.clojure
-        - match: '(?=\])'
-          pop: true
-        - match: '(?<=\{)'
-          comment: "TODO: merge with map"
-          push:
-            - meta_scope: meta.expression.map.clojure
-            - match: "}"
-              scope: punctuation.definition.map.end.clojure
-              pop: true
-            - include: all
-        - include: binding
-    - match: '(?=#\{)'
-      push:
-        - meta_scope: meta.structure.binding_exp.set.clojure
-        - match: '(?=\])'
-          pop: true
-        - include: set
-        - include: binding
-    - match: (?=")|(?=\\)|(?=\:)|(?=\#")
-      push:
-        - meta_scope: meta.structure.binding_exp.string.clojure
-        - match: '(?=\])'
-          pop: true
-        - include: string
-        - include: binding
-    - match: '(?=[a-zA-Z+!\-_?0-9*~#@''`/.$=])'
-      comment: symbol matching
-      push:
-        - meta_scope: meta.structure.binding_exp.symbol.clojure
-        - match: '(?=\])'
-          pop: true
-        - match: '(?=[a-zA-Z+!\-_?0-9*~#@''`/.$=])'
-          push:
-            - match: '(?<=[a-zA-Z+!\-_?0-9*~#@''`/.$=])(?![a-zA-Z+!\-_?0-9*~#@''`/.$=])'
-              pop: true
-            - include: keyword
-            - include: operator
-            - include: number
-            - include: symbol
-        - match: '(?![a-zA-Z+!\-_?0-9*~#@''`/.$=])'
-          push:
-            - match: '(?=\])'
-              pop: true
-            - include: binding
-    - match: '[^\s]'
-      push:
-        - meta_scope: invalid.illegal.bindings.clojure
-        - match: '[^\]]'
-          pop: true
-  bindings_form:
+    - include: match_expr
+  match_expr:
+    - include: match_noise
+    - match: '[)\]}]'
+      scope: invalid.illegal.clojure
+    - match: '\('
```

Comment:
You can omit the redundant single quotes around this regex, and various other regexes. Download PackageDev and enable the "Sublime Syntax" syntax from that package to better see what is possible and what is not possible within regexes. I highly recommend that package.